docs/root/intro/arch_overview/other_features/compression.rst

```diff
@@ -0,0 +1,20 @@
+.. _arch_overview_compression:
+
+Compression
+===========
+
+Underlying implementation
+-------------------------
+
+Currently Envoy is written to use `zlib <http://zlib.net>`_ as the compression provider.
+
+.. note::
+
+  `zlib-ng <https://github.com/zlib-ng/zlib-ng>`_ is a fork that hosts several 3rd-party
+  contributions containing new optimizations. Those optimizations are considered useful for
+  `improving compression performance <https://github.com/envoyproxy/envoy/issues/8448#issuecomment-667152013>`_.
+  Envoy can be built with `zlib-ng <https://github.com/zlib-ng/zlib-ng>`_ support by using
+  ``--define zlib=ng`` Bazel option. In order to use `zlib-ng <https://github.com/zlib-ng/zlib-ng>`_
+  with optimization turned on, apply ``--define zlib=ng-with-optimization``. Please note that
```

Comment:
optimizations

Reply:
Should the flag changed to `zlib=ng-with-optimizations` (with `s`) as well?

Reply:
Yeah probs